src/output.rs

```diff
@@ -816,6 +817,45 @@ impl<W: io::Write> Formatter<W> for Summary {
         )
     }
 
+    fn produce_header(
+        metrics: &Metrics,
+        mut line: impl FnMut(fmt::Arguments) -> Result<(), io::Error>
+    ) -> Result<(), io::Error> {
+        line(format_args!("Summary at {}", metrics.time))?;
+        for tal in &metrics.tals {
+            line(format_args!(
+                "{}: {} verified ROAs, {} verified VRPs, \
+                 {} unsafe VRPs, {} final VRPs.",
+                tal.name(), tal.publication.valid_roas, tal.vrps.valid,
+                tal.vrps.marked_unsafe, tal.vrps.contributed
+            ))?;
+        }
+        line(format_args!(
+            "total: {} verified ROAs, {} verified VRPs, \
+             {} unsafe VRPs, {} final VRPs.",
+            metrics.publication.valid_roas,
+            metrics.vrps.valid, metrics.vrps.marked_unsafe,
+            metrics.vrps.contributed,
+        ))
+    }
+
+    pub fn log(metrics: &Metrics) {
+        Self::produce_header(metrics, |args| {
+            info!("{}", args);
+            Ok(())
+        }).unwrap()
```

Comment:
If for some very exotic reason we can not write to stdout (or, to wherever stdout is redirected), this `unwrap` can cause trouble I guess?

Reply:
The only way that `produce_header` can return an error is if the closure returns an error, which, in this case it never does, so this is fine.